src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1019,31 +1020,31 @@ def _get_auth_provider_latest_api_version(cli_ctx):
     return api_version
 
 
-def _update_provider(cli_ctx, namespace, registering, wait, mg_id=None, accept_terms=None):
+def _update_provider(cli_ctx, namespace, registering, wait, properties=None, mg_id=None, accept_terms=None):
     import time
     target_state = 'Registered' if registering else 'Unregistered'
-    rcf = _resource_client_factory(cli_ctx)
+    client = cf_providers(cli_ctx, None, api_version='2021-04-01')
```

Comment:
May I ask why you do not upgrade the api-version for `ResourceType.MGMT_RESOURCE_RESOURCES`, but specify a fixed version `2021-04-01` here?
Actually, If the current api-version does not support new features, we will generally bump the api-version